fetch_binary_drivers/package.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>fetch_drivers</name>
+  <version>0.0.0</version>
+  <description>
+    The public fetch_drivers package is a binary only release.
+
+    fetch_drivers contains both the drivers and firmware for the fetch and freight research robots.
+    There should be no reason to use these drivers unless you're running on a fetch or a frieght research robot.
+    This package, is a cmake/make only package which installs the binaries for the drivers and firmware.
+  </description>
+
+  <!-- One maintainer tag required, multiple allowed, one person per tag -->
+  <maintainer email="amoriarty@fetchrobotics.com">Alexander Moriarty</maintainer>
+  <maintainer email="opensource@fetchrobotics.com">FetchRobotics Open Source Team</maintainer>
+
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <license>Proprietary</license>
+
+
+  <!-- Url tags are optional, but multiple are allowed, one per tag -->
+  <url type="website">http://wiki.ros.org/fetch_drivers</url>
+  <url type="docs">https://docs.fetchrobotics.com</url>
+  <url type="info">https://fetchrobotics.com/robotics-platforms/</url>
+
+
+  <!-- Author tags are optional, multiple are allowed, one per tag -->
+  <author email="amoriarty@fetchrobotics.com">Alexander Moriarty</author>
+
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <!-- mk is actually the only build time depend; it just pulls down a the compiled objects. -->
+  <build_depend>mk</build_depend>
+
+  <!-- system dependencies which the drivers dyanmically link against -->
+  <exec_depend>boost</exec_depend>
+  <exec_depend>curl</exec_depend>
+  <exec_depend>python</exec_depend>
+  <exec_depend>yaml-cpp</exec_depend>
+
+  <!-- ros msgs which the drivers depend on -->
+  <exec_depend>actionlib_msgs</exec_depend>
+  <exec_depend>diagnostic_msgs</exec_depend>
+  <exec_depend>fetch_driver_msgs</exec_depend>
+  <exec_depend>fetch_auto_dock_msgs</exec_depend>
+  <exec_depend>nav_msgs</exec_depend>
+  <exec_depend>power_msgs</exec_depend>
+  <exec_depend>robot_calibration_msgs</exec_depend>
```

Comment:
you won't be able to do that -- the ROS buildfarm / repos only contain the latest debs - once a new version is synced, the old ones go away. Hardcoding the value would make your package uninstallable at best.

Reply:
@mikeferguson: would a nightly build job of our binaries, ensuring we’ve always got the latest dependencies work here?

@cjds and I were hoping to open source the Dockerfile & Python script which builds the `output.tar.gz`- it won’t be helpful without pull access on our drivers but it would at least make the process transparent

Reply:
@moriarty even having the released deb download a nightly binary build is a challenge -- when a sync to public happens, you'd have a ~24 hour window where the version built against public is no longer right. (and that sync happens every ~ 2 weeks, so that's about 7% downtime). Also how do you make it work on both shadow-fixed (where developers tend to test) and the main ros repo?

Reply:
:disappointed:
@dbking77 & @chadrockey maybe long term we reopen the discussion about refactoring some things to better support public/private drivers to make releasing to research customers easier :question: 

We might need a second version setup up and built against shadow-fixed, this still doesn't fix the original problem, and it also wouldn't be clear how to set that up so that the shadow-fixed pulled a `fetch-drivers-shadow.tar.gz` and the release pulled `fetch-drivers.tar.gz` especially since currently they're pulling: `http://packages.fetchrobotics.com/binaries/fetch-drivers-0.8.0.tar.gz`
... Fetch3 & my machine are currently setup against shadow-fixed, but these drivers were built against whatever is inside of

```dockerfile
FROM ros:melodic-robot-bionic
```
(As mentioned @cjds will cleanup the creation process and we plan to put them:
https://github.com/fetchrobotics/fetch_firmware_repackager or rename that repository `fetch_drivers_repackager`)

And it works now, but that's likely because there hasn't been any API/ABI breaking changes, which is not always guaranteed.
We'll need to check which of the dependencies we have are core dependencies, and which are non-core dependencies. 

For the most part, the 7% downtime is a worst case calculation coming from 24hr every 2 weeks. If the dependencies are part of the core ros repositories, I think the API/ABI is stable enough or at least suppose to be... For non-core, and not released by us then it's just a risk that we'll probably need to take.
I don't even know if the melodic sync is triggered every 2 weeks because most of the folks at open robots are developing towards ros2. @jacobperron & @wjwwood?

Our plan, was to setup a mirror of ROS, which we manually trigger a sync on.
So customers with Fetch Research Robots could point to our http://packages.fetchrobotics.com/ and get guaranteed stability. But have the option to get the packages from packages.ros.org either ros or ros-shadow-fixed if they really want to live on the edge and accept some instability.

By pushing this into the public build farm, our internal `packages.fetchrobotics.com` will become almost a pure mirror, which should have much maintainable for our DevOps team and @bluryi.

If we sync our mirror manually, after testing to know we're not in the 7% downtime, then we'll get stability and more frequent updates.

For the 7% of possible down time for the `packages.ros.org` I think @erelson will need to update the (not yet existing) documentation to include strong warnings about pointing to `packages.fetchrobotics.com` or `packages.ros.org`
The packages.fetchrobotics.com mirror for *melodic* does not yet exist, we were going to quietly get our packages into melodic, and then setup a mirror once they were in and synced.

Reply:
@clalancette is the ROS Boss for Melodic, so it's up to him to decide when the syncs happen, though I think they've been trying to keep a regular cadence. Looking at the discourse posts, it's more like once a month, though maybe that will change.

Perhaps you can anticipate the syncs and be prepared with updated packages of your own to sync in order to minimize the period where your debs are out of date. We do usually announce an intention to sync a week or two before doing it, with the intention that people could test it for unintended regressions.

----

I don't know if this would be relevant as well, but there's this recent announcement by @nuclearsandwich:

https://discourse.ros.org/t/announcing-ros-snapshot-repositories/7705

Reply:
My intention is to sync Melodic every two weeks, though I've basically never been able to achieve that due to regressions that I have to fix for every sync.  So the cadence has indeed been more like once a month.

I intend to continue to attempt to sync every two weeks unless there is a compelling reason to change that.  So if you have reasons you'd like it to change, please let me know and we can consider it.

Reply:
@nickswalker and @mikeferguson I wanted to give you a warning: just in case you've been running melodic and testing with Fetch's using **ros-shadow-fixed** this is now there:

http://repositories.ros.org/status_page/ros_melodic_default.html?q=fetch_

@erelson will do a test on hardware using the **ros-shadow-fixed** before we make an official announcement with links to upgrade documentation. So: best case, everything works, but it's more likely that the official announcement to the fetch user mailing lists is going to be not the upcoming ros melodic sync, but the one after. If we're really unlucky we could be in one of those 7% downtime scenarios which was mentioned above.